arex-web-core/src/main/java/com/arextest/web/core/repository/mongo/ReplayCompareResultRepositoryImpl.java

```diff
@@ -197,6 +197,32 @@ public int queryCompareResultCountByPlanId(String planId) {
         return mongoTemplate.findDistinct(query, RECORD_ID, ReplayCompareResultCollection.class, String.class).size();
     }
 
+    @Override
+    public List<CompareResultDto> queryFailCompareResults(String planId,
+                                                          List<String> planItemIdList,
+                                                          List<String> recordIdList,
+                                                          List<Integer> diffResultCodeList) {
+        Query query = Query.query(
+                Criteria.where(PLAN_ID).is(planId)
```

Comment:
The 'planId' field needs to be robustly judged or use @NotNull to attribute the 'planId' in QueryPlanFailCaseRequestType